Examples/Python/src/Geometry.cpp

```diff
@@ -96,16 +121,200 @@ void addGeometry(Context& ctx) {
           return hook;
         }));
   }
+}
+
+void addExperimentaGeometry(Context& ctx) {
+  auto [m, mex] = ctx.get("main", "examples");
+
+  using namespace Acts::Experimental;
+
+  {  // Detector definition
+    py::class_<Detector, std::shared_ptr<Detector>>(m, "Detector");
+  }
+
+  {  // Detector volume definition
+    py::class_<DetectorVolume, std::shared_ptr<DetectorVolume>>(
+        m, "DetectorVolume");
+  }
 
   {
-    py::class_<Acts::Experimental::Detector,
-               std::shared_ptr<Acts::Experimental::Detector>>(m, "Detector");
+    // The surface hiearchy map
+    using SurfaceHierarchyMap =
+        Acts::GeometryHierarchyMap<std::shared_ptr<Surface>>;
+
+    py::class_<SurfaceHierarchyMap, std::shared_ptr<SurfaceHierarchyMap>>(
+        m, "SurfaceHierarchyMap");
+
+    // Extract volume / layer surfaces
+    mex.def("extractVolumeLayerSurfaces", [](const SurfaceHierarchyMap& smap,
+                                             bool sensitiveOnly) {
+      std::map<unsigned int,
+               std::map<unsigned int, std::vector<std::shared_ptr<Surface>>>>
+          surfaceVolumeLayerMap;
+      for (const auto& surface : smap) {
+        auto gid = surface->geometryId();
+        // Exclusion criteria
+        if (sensitiveOnly and gid.sensitive() == 0) {
+          continue;
+        };
+        surfaceVolumeLayerMap[gid.volume()][gid.layer()].push_back(surface);
+      }
+      // Return the surface volume map
+      return surfaceVolumeLayerMap;
+    });
   }
 
   {
-    py::class_<Acts::Experimental::DetectorVolume,
-               std::shared_ptr<Acts::Experimental::DetectorVolume>>(
-        m, "DetectorVolume");
+    // Be able to construct a proto binning
+    auto pBinning =
+        py::class_<ProtoBinning>(m, "ProtoBinning")
+            .def(py::init<Acts::BinningValue, Acts::detail::AxisBoundaryType,
+                          const std::vector<Acts::ActsScalar>&, std::size_t>())
+            .def(py::init<Acts::BinningValue, Acts::detail::AxisBoundaryType,
+                          Acts::ActsScalar, Acts::ActsScalar, std::size_t,
+                          std::size_t>());
```

Comment:
the block can be removed from this one

```suggestion
  // Be able to construct a proto binning
  py::class_<ProtoBinning>(m, "ProtoBinning")
          .def(py::init<Acts::BinningValue, Acts::detail::AxisBoundaryType,
                        const std::vector<Acts::ActsScalar>&, std::size_t>())
          .def(py::init<Acts::BinningValue, Acts::detail::AxisBoundaryType,
                        Acts::ActsScalar, Acts::ActsScalar, std::size_t,
                        std::size_t>());
```

Reply:
Cool.